### PR TITLE
DBT Platform Migration

### PR DESCRIPTION
**Pull request by acodeninja**
### Aim

Prepare the `lite-api` codebase for migration to DBT platform from GOV PaaS.

[DPM-1283](https://uktrade.atlassian.net/browse/DPM-1283)

[DPM-1283]: https://uktrade.atlassian.net/browse/DPM-1283?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by depsiatwal** [comment]
Tested on dev2 deployment is ok.

**Review by markj0hnst0n — APPROVED** [review]

**Review by hnryjmes — APPROVED** [review]
approving but I'd note that @currycoder is still reviewing this

**Review by currycoder — COMMENTED** [review]
I've missed the opportunity to comment before the review was merged, but would appreciate some discussions on the comments that I've opened if only for my understanding.  @acodeninja

File: api/documents/libraries/s3_operations.py [hunk 1]
```diff
@@ -23,14 +24,23 @@ def init_s3_client():
...
-        **additional_s3_params,
-    )
+
+    if is_copilot():
+        _client = boto3.client(
+            "s3",
+            region_name=settings.AWS_REGION,
+            config=Config(connect_timeout=settings.S3_CONNECT_TIMEOUT, read_timeout=settings.S3_REQUEST_TIMEOUT),
+            **additional_s3_params,
+        )
```
[the remark this answers is not in the document]
Comment (depsiatwal) [reply api/documents/libraries/s3_operations.py:34]: They will be walking us though that but it's done via DBT platform AWS pilot config

File: api/conf/settings.py [hunk 2]
```diff
@@ -141,6 +137,14 @@
     "drf_spectacular",
 ]
 
+if not IS_ENV_DBT_PLATFORM:
+    INSTALLED_APPS += [
+        "health_check.db",
+        "health_check.cache",
+        "health_check.storage",
+        "health_check.contrib.migrations",
+    ]
```
[the remark this answers is not in the document]
Comment (depsiatwal) [reply api/conf/settings.py:146]: For know there's a know issue with these health check on DBT platform. 
we will look at that independently

File: .copilot/phases/pre_build.sh [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+
+cat <<EOF > ./.gitmodules
+[submodule "lite-content"]
+path = lite_content
+url = $git_clone_base_url/lite-content.git
+branch = master
+[submodule "lite_routing"]
+path = lite_routing
+url = $git_clone_base_url/lite-routing.git
+branch = main
```
[the remark this answers is not in the document]
Comment (depsiatwal) [reply .copilot/phases/pre_build.sh:19]: Jira raised and DBT Platform team will be looking at it 
https://uktrade.atlassian.net/browse/LTD-5627

File: api/documents/libraries/s3_operations.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (acodeninja) [reply api/documents/libraries/s3_operations.py:34]: In DBT platform ECS tasks have access to a metadata endpoint, allowing them to obtain credentials for the IAM role assigned to the task, see [configuring credentials](https://boto3.amazonaws.com/v1/documentation/api/latest/guide/credentials.html#configuring-credentials) point 10 (metadata endpoint).

File: api/conf/settings.py [hunk 4]
```diff
@@ -321,6 +325,7 @@
         environment=env.str("SENTRY_ENVIRONMENT"),
         integrations=[DjangoIntegration()],
         send_default_pii=True,
+        traces_sample_rate=env.float("SENTRY_TRACES_SAMPLE_RATE", 1.0),
```
[the remark this answers is not in the document]
Comment (acodeninja) [reply api/conf/settings.py:328]: Yes, 100% tracing in prod will cost too much / take down self hosted sentry.

File: .copilot/phases/pre_build.sh (hunk 3 above)
[the remark this answers is not in the document]
Comment (kevincarrogan) [reply .copilot/phases/pre_build.sh:19]: This does all feel a little odd.

I'm presuming that this is some weirdness where DBT Platform isn't actually cloning the repo properly and this is a way or re-injecting back in the submodules.

[the remark this answers is not in the document]
Comment (acodeninja) [reply .copilot/phases/pre_build.sh:19]: We're using the same `.gitmodules` file from https://github.com/uktrade/lite-api/blob/v1.177.0/.gitmodules but we inject a different url to connect to GitHub via AWS Codestar. The existing behaviour for checking out the submodules at a specific commit SHA remains intact.

If we look at the `dev` branch, we see the following on GitHub:

[django_db_anonymiser @ eaed065](https://github.com/uktrade/django-db-anonymiser/tree/eaed065b670cceaa6564053277dae242c72e7b90)
[lite_content @ 45c6d3b](https://github.com/uktrade/lite-content/tree/45c6d3b96ce3db3f59f629683b254f59bc92448f)
[lite_routing @ bf87c84](https://github.com/uktrade/lite-routing/tree/bf87c8492ae0a7dee8b28c577ffb7d1f62db94f4)

And then in the build job we see:

```console
Cloning into '/codebuild/output/src3936388621/src/github.com/uktrade/lite-api/django_db_anonymiser'...
Cloning into '/codebuild/output/src3936388621/src/github.com/uktrade/lite-api/lite_content'...
Cloning into '/codebuild/output/src3936388621/src/github.com/uktrade/lite-api/lite_routing'...
Submodule path 'django_db_anonymiser': checked out 'eaed065b670cceaa6564053277dae242c72e7b90'
Submodule path 'lite_content': checked out '45c6d3b96ce3db3f59f629683b254f59bc92448f'
Submodule path 'lite_routing': checked out 'bf87c8492ae0a7dee8b28c577ffb7d1f62db94f4'
```

Which matches the above commit SHAs from GitHub.

**Pull request merged**